nixos-modules/host.nix

```
@@ -224,6 +224,18 @@ in
         restartTriggers = [guestConfig.system.build.toplevel];
         overrideStrategy = "asDropin";
       };
+      "microvm-tap-interfaces@${name}" = {
+        inherit (microvmConfig) restartIfChanged;
+        overrideStrategy = "asDropin";
+      };
+      "microvm-pci-devices@${name}" = {
+        inherit (microvmConfig) restartIfChanged;
+        overrideStrategy = "asDropin";
+      };
+      "microvm-virtiofsd@${name}" = {
+        inherit (microvmConfig) restartIfChanged;
```

```suggestion
        restartIfChanged = [ "" inherit (microvmConfig) restartIfChanged];
        overrideStrategy = "asDropin";
      };
      "microvm-pci-devices@${name}" = {
        restartIfChanged = [ "" inherit (microvmConfig) restartIfChanged];
        overrideStrategy = "asDropin";
      };
      "microvm-virtiofsd@${name}" = {
        restartIfChanged = [ "" inherit (microvmConfig) restartIfChanged];
```
I don't really have something to test this without causing downtimes for others but shouldn't this clear the value of the template and make your usecase work again?

---

You have to read the upstream nixpkgs source I linked before. You will see that `restartIfChanged` is not an option in `serviceConfig` and thus cannot be used like you are using it. I believe you should use `serviceConfig."X-RestartIfChanged"` instead.

---

good point. Not having a good one with this PR 😂